File: MetaServer/project/static/javascript/modelrectangle.js
function ModelRectangle(modelPoint1, modelPoint2) {
	this.min_x;
	this.max_x;
	this.min_y;
	this.max_y;

	if (modelPoint1.x < modelPoint2.x)
	{
		this.min_x = modelPoint1.x;
		this.max_x = modelPoint2.x;
	}
	else
	{
		this.min_x = modelPoint2.x;
		this.max_x = modelPoint1.x;
	}
	if (modelPoint1.y < modelPoint2.y)
	{
		this.min_y = modelPoint1.y;
		this.max_y = modelPoint2.y;
	}
	else
	{
		this.min_y = modelPoint2.y;
		this.max_y = modelPoint1.y;
	}

	this.Contains = function(modelPoint) {
		return (this.min_x <= modelPoint.x && modelPoint.x <= this.max_x && this.min_y <= modelPoint.y && modelPoint.y <= this.max_y);
	};
}


export {ModelRectangle};

File: Editor/Editor/ModelRectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Editor
{
    class ModelRectangle
    {
        public double min_x;
        public double max_x;
        public double min_y;
        public double max_y;

        public ModelRectangle(ModelPoint p1, ModelPoint p2)
        {
            if (p1.x < p2.x)
            {
                min_x = p1.x;
                max_x = p2.x;
            }
            else
            {
                min_x = p2.x;
                max_x = p1.x;
            }
            if (p1.y < p2.y)
            {
                min_y = p1.y;
                max_y = p2.y;
            }
            else
            {
                min_y = p2.y;
                max_y = p1.y;
            }
        }

        public bool contains(ModelPoint p)
        {
            return (min_x <= p.x && p.x <= max_x && min_y <= p.y && p.y <= max_y);
        }
    }
}
